client/src/pages/Login.js

```diff
@@ -29,21 +28,17 @@ function Login(props: Props) {
       form.validateFields((err, values) => {
         if (!err) {
           const { email, password } = values;
-          login({ email, password }).then(res => {
-            if (res.status === 200) {
-              localStorage.setItem('token', res.token);
-
-              setAuthed(true);
-              setAuthRole(res.permission);
-              setError('');
+          login({ email, password }).then(errorMessage => {
+            if (errorMessage !== null) {
+              setError(errorMessage);
             } else {
-              setError(res.message);
+              setError('');
```

Comment:
We should have a default here just so we know what goes wrong. If it shouldn't trigger then we can just remove the conditional statements all together.

Reply:
I'll change it to reset to set to null, hopefully making it more clear that there's no error.  In the case where you try to log in, get an error, then successfully log in, we want to set the error to be blank, which is why we need the conditional